shoov/modules/custom/shoov_ui_build/drush/shoov_ui_build.drush.inc

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/**
+ * @file
+ * Drush commands for UI Builds.
+ */
+
+/**
+ * Implements hook_drush_command().
+ */
+function shoov_ui_build_drush_command() {
+  $items = array();
+
+  $items['delete-old-ui-builds'] = array(
+    'description' => 'Delete old UI builds and screenshots.',
+  );
+
+  return $items;
+}
+
+/**
+ * Command callback for "delete-old-ui-builds".
+ *
+ * Deletes ui-builds that are older than one month, meanwhile keeping 10 builds
+ * in each repository.
+ */
+function drush_shoov_ui_build_delete_old_ui_builds() {
+
+  $query = new EntityFieldQuery();
+  $result = $query
+    ->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'repository')
+    ->propertyCondition('status', NODE_PUBLISHED)
+    ->propertyOrderBy('nid', 'ASC')
+    ->execute();
+
+  if (empty($result['node'])) {
+    // Stop if there's not repositories.
+    drush_print('No repositories found.');
+    return;
+  }
+
+  $ids = array_keys($result['node']);
+  $repos = node_load_multiple($ids);
+  foreach ($repos as $repo) {
+    drush_print('Checking Repository "' . $repo->title . '"');
+
+    // Need to use db_select() because field condition override in OG doesn't
+    // work in a Drush command.
+    $ui_build_query = db_select('og_membership', 'm');
+    $ui_build_query->leftJoin('node', 'n', 'm.etid = n.nid');
+    $ui_build_results = $ui_build_query->fields('n', array('nid'))
+      ->condition('m.gid', $repo->nid)
+      ->condition('m.entity_type', 'node')
+      ->condition('n.type', 'ui_build')
+      ->execute()
+      ->fetchAll();
+
+    if (count($ui_build_results) < 10) {
+      // Don't delete anything if we have less than 10 builds in this repository.
+      drush_print('There is no more than 10 builds in this repository.');
+      continue;
+    }
+
+    // Delete every build that is older than 30 days.
+    foreach ($ui_build_results as $key => $ui_build) {
+      $ui_build_wrapper = entity_metadata_wrapper('node', $ui_build->nid);
+
+      if ($ui_build_wrapper->created->value() > strtotime('-1 month') || count($ui_build_results) < 10) {
```

Comment:
Didn't understand the logic first. 
So if you order them by creation date ASC, once you get the build that is younger than 30 days or count of left builds becomes less than 10 - you will continue.